spec/lib/oauth/helpers/resource_indicators_checker_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Doorkeeper::OAuth::Helpers::ResourceIndicatorsChecker do
+  describe ".valid?" do
+    before do
+      allow(Doorkeeper.config).to receive(:using_resource_indicators?).and_return(true)
+    end
+    
+    let(:subject) do
+      described_class.valid?(
+        double(resource_indicators: Doorkeeper::OAuth::ResourceIndicators.from_array(preapproved_indicators)),
+        Doorkeeper::OAuth::ResourceIndicators.from_array(requested_indicators),
+      )
+    end
+
+    context "when preapproved is a superset" do
+      let(:preapproved_indicators) do
+        ["http://example.com/1", "http://example.com/2"]
```

Comment:
Is it worth including a case for exact match and/or duplicates resource indicators?

Reply:
yeah don't see why not